File: Python/Arrays&Hashing/LongestConsecutiveSequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        # Solution 1
        '''
        if len(nums)<2:
            return len(nums)
        temp = res = 1
        nums = sorted(set(nums))
        for i, numi in enumerate(nums[:-1:]):
            if numi + 1 == nums[i+1]:
                temp+=1
            else:
                temp=1
            if temp > res:
                res = temp
        return res
        '''
        # Solution 2
        numSet = set(nums)
        longest = 0

        for n in nums:
            if (n-1) not in numSet:
                length = 0
                while (n + length) in numSet:
                    length += 1
                longest = max(length, longest)
        return longest

File: JavaScript/Arrays&Hashing/LongestConsecutiveSequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    const numSet = new Set(nums.sort((a, b) => a - b));
    let longestSeq = 0;
    for (let num of numSet) {
        if (!numSet.has(num - 1)) {
            let currentSeq = 1;
            while (numSet.has(num + currentSeq)) currentSeq++;
            longestSeq = Math.max(longestSeq, currentSeq);
        }
    }
    return longestSeq;
};
